### PR TITLE
add com.obsproject.Studio.Plugin.Vaapi

### PR DESCRIPTION
**Pull request by RushingAlien**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** : https://github.com/fzwoch/obs-vaapi/issues/8
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

https://github.com/flathub/flathub/pull/3862 : Previous discussions and reviews before i screwed up and accidentally closed the pr by rebasing, sorry for the inconvenience

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build com.obsproject.Studio.Plugin.Vaapi

**Comment by tytan652** [comment]
For the 0.2.0 version, you will need to add libdir option like it has been done for obs-gstreamer.

https://github.com/flathub/com.obsproject.Studio.Plugin.Gstreamer/blob/master/com.obsproject.Studio.Plugin.Gstreamer.yaml#L11

``` yaml
build-options:
    prefix: /app/plugins/Vaapi
  libdir: lib/obs-plugins
```

**Comment by RushingAlien** [comment]
Thanks for the heads up @tytan652, will do

**Comment by barthalion** [comment]
cc @GeorgesStavracas

File: com.obsproject.Studio.Plugin.Vaapi.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	<name>Vaapi Plugin</name>
+	<summary>GStreamer based VAAPI encoder implementation.</summary>
+	<url type="homepage">https://github.com/fzwoch/obs-vaapi</url>
+	<releases>
+		<release version="0.2.0" date="2022-09-11"/>
+	</releases>
+	<metadata_license>CC0-1.0</metadata_license>
+	<project_license>GPL-2.0</project_license>
+</component>
+
```
Comment (GeorgesStavracas) [inline com.obsproject.Studio.Plugin.Vaapi.metainfo.xml:14]: Please use 2 spaces instead of tabs for indentation

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+	"skip-icons-check": true,
+	"only-arches": [ "x86_64" ]
+}
+
```
Comment (GeorgesStavracas) [inline flathub.json:5]: Here too

File: com.obsproject.Studio.Plugin.Vaapi.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+	<id>com.obsproject.Studio.Plugin.Vaapi</id>
+	<extends>com.obsproject.Studio</extends>
+	<name>Vaapi Plugin</name>
+	<summary>GStreamer based VAAPI encoder implementation.</summary>
```
Comment (GeorgesStavracas) [inline com.obsproject.Studio.Plugin.Vaapi.metainfo.xml:6]: `GStreamer-based VA-API encoder`

(Descriptions don't have a ending dot)

File: com.obsproject.Studio.Plugin.Vaapi.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+	<id>com.obsproject.Studio.Plugin.Vaapi</id>
+	<extends>com.obsproject.Studio</extends>
+	<name>Vaapi Plugin</name>
```
Comment (GeorgesStavracas) [inline com.obsproject.Studio.Plugin.Vaapi.metainfo.xml:5]: `GStreamer VA-API Plugin`

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]

File: com.obsproject.Studio.Plugin.Vaapi.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+id: com.obsproject.Studio.Plugin.Vaapi
+branch: stable
+runtime: com.obsproject.Studio
+runtime-version: stable
+sdk: org.kde.Sdk//6.4
+build-extension: true
+separate-locales: false
+appstream-compose: false
+build-options:
+  prefix: /app/plugins/Vaapi
```
Comment (GeorgesStavracas) [inline com.obsproject.Studio.Plugin.Vaapi.yml:10]: `/app/plugins/GStreamerVaapi`

File: com.obsproject.Studio.Plugin.Vaapi.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+id: com.obsproject.Studio.Plugin.Vaapi
```
Comment (GeorgesStavracas) [inline com.obsproject.Studio.Plugin.Vaapi.yml:1]: `com.obsproject.Studio.Plugin.GStreamerVaapi`

File: com.obsproject.Studio.Plugin.Vaapi.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>com.obsproject.Studio.Plugin.Vaapi</id>
```
Comment (GeorgesStavracas) [inline com.obsproject.Studio.Plugin.Vaapi.metainfo.xml:3]: `com.obsproject.Studio.Plugin.GStreamerVaapi`

File: com.obsproject.Studio.Plugin.Vaapi.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+runtime-version: stable
+sdk: org.kde.Sdk//6.4
+build-extension: true
+separate-locales: false
+appstream-compose: false
+build-options:
+  prefix: /app/plugins/Vaapi
+  libdir: lib/obs-plugins
+modules:
+  - name: vaapi
```
Comment (GeorgesStavracas) [inline com.obsproject.Studio.Plugin.Vaapi.yml:13]: `gstreamer-vaapi`

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]
I think it might be safer to use `GStreamerVaapi` around, because that's technically what the plugin is

**Comment by GeorgesStavracas** [comment]
(Please rename the files to `com.obsproject.Studio.Plugin.GStreamerVaapi` too)

**Comment by RushingAlien** [comment]
Is it fine if i don't change the branch name?

File: README.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# com.obsproject.Studio.Plugin.Vaapi
+Pull requests for new applications to be added
+to install and test this OBS plugin simply run 
+```
+flatpak install --user https://dl.flathub.org/build-repo/2496/com.obsproject.Studio.Plugin.Vaapi.flatpakref
+```
+make sure to also install `org.freedesktop.Platform.GStreamer.gstreamer-vaapi` as it is a required package for this plugin to work
+```
+flatpak install org.freedesktop.Platform.GStreamer.gstreamer-vaapi//22.08
+```
```
Comment (GeorgesStavracas) [inline README.md:10]: You might need to update these too

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]
One last comment. LGTM after that

File: README.md (hunk 9 above)
Comment (RushingAlien) [reply README.md]: You mean the `h1` of the README?

Comment (GeorgesStavracas) [reply README.md]: Yup

File: README.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# com.obsproject.Studio.Plugin.GStreamerVaapi
+Pull requests for new applications to be added
+
+To install and test this OBS plugin simply run:
+```
+flatpak install --user https://dl.flathub.org/build-repo/2496/com.obsproject.Studio.Plugin.Vaapi.flatpakref
```
Comment (GeorgesStavracas) [inline README.md:6]: Here too

Comment (RushingAlien) [reply README.md:6]: Do I change it to as if it is already on flathub? as in 
```
flatpak install com.obsproject.Studio.Plugin.GStreamerVaapi
```
?

Comment (GeorgesStavracas) [reply README.md:6]: Sounds good to me! Please change to that

**Review by GeorgesStavracas — APPROVED** [review]

**Pull request closed without merging**